src/cls/tabular/fbreader_fbu2.cc

```diff
@@ -0,0 +1,321 @@
+            std::cout << float_col_data->data()->Get(0) ;
+            break ;
+          }
+          case Tables::DataTypes_FBU_SDT_STRING_FBU : {
+            auto string_col_data = static_cast< const Tables::SDT_STRING_FBU* >( curr_rec_data->Get(j) ) ;
+            std::cout << string_col_data->data()->Get(0)->str() ;
+            break ;
+          }
+          default :
+            std::cout << "execute_query: unrecognized row_data_type "
+                      << (unsigned)curr_rec_data_type->Get(j) << std::endl ;
+            exit(1) ;
+        } //switch
+
+        if( j < curr_rec_data->Length()-1 )
+          std::cout << "," ;
+
+      } //for loop
+      std::cout << std::endl ;
+    } //for loop
+  } // Relation_Rows
+
+  // process one Root>Cols flatbuffer
+  else if( format == Tables::Relation_FBU_Cols_FBU ) {
+    if( print_verbose )
+      std::cout << "else if data_type == Tables::Relation_FBU_Cols_FBU" << std::endl ;
+
+    auto cols = static_cast< const Tables::Cols_FBU* >( data ) ;
+    auto cols_data = cols->data() ;
+
+    // collect data for stdout printing
+    std::vector< std::vector< std::string > > out_data ;
+    for( unsigned int i = 0; i < cols_data->Length(); i++ ) {
+      std::vector< std::string > empty_vect ;
+      out_data.push_back( empty_vect ) ;
+    }
+
+    std::vector< uint64_t > rids ;
+    for( unsigned int i = 0; i < cols_data->Length(); i++ ) {
+
+      auto col = static_cast< const Tables::Col_FBU* >( cols_data->Get(i) ) ;
+      auto col_rids      = col->RIDs() ;
+      auto col_data      = col->data() ;
+      auto col_data_type = col->data_type() ;
+
+      if( i == 0 ) {
+        for( unsigned int k = 0; k < col_rids->Length(); k++ )
+          rids.push_back( col_rids->Get(k) ) ;
+      }
+
+      for( unsigned int j = 0; j < nrows; j++ ) {
+        switch( (unsigned)col_data_type ) {
+          case Tables::DataTypes_FBU_SDT_UINT64_FBU : {
+            auto int_col_data = static_cast< const Tables::SDT_UINT64_FBU* >( col_data ) ;
+            out_data[i].push_back( std::to_string( int_col_data->data()->Get(j) ) ) ;
+            break ;
+          }
+          case Tables::DataTypes_FBU_SDT_FLOAT_FBU : {
+            auto float_col_data = static_cast< const Tables::SDT_FLOAT_FBU* >( col_data ) ;
+            out_data[i].push_back( std::to_string( float_col_data->data()->Get(j) ) ) ;
+            break ;
+          }
+          case Tables::DataTypes_FBU_SDT_STRING_FBU : {
+            auto string_col_data = static_cast< const Tables::SDT_STRING_FBU* >( col_data ) ;
+            out_data[i].push_back( string_col_data->data()->Get(j)->str() ) ;
+            break ;
+          }
+          default :
+            std::cout << "unrecognized data_type " << (unsigned)col_data_type << std::endl ;
+            exit(1) ;
+        } //switch
+      } //for
+    } //for
+
+    for( unsigned int i = 0; i < nrows; i++ ) {
```

Comment:
I think this method will print a bit differently than the other AsCsv() methods, please check the flatflex one in particular.  A check is needed for print verbose or not, in which case you can print the entire skyrec header not just the rid as below.  Also this_row + "," should use the delim like the other functions do, I think this defaults to "|", and I believe no begin/trail delim but please check the output of flatflex print.  thanks

Reply:
fbwriter_fbu2 is designed to be a stand-alone code capable of reading fbu data. I use it as a reference for nailing down the read logic before adopting the tabular abstractions or  making integration changes to cls_tabular files. 
it's purely a development tool and not part of the skyhook core.
switched out the comma for a pipe.
would it be ok to keep it in the repo in case of schema changes?

Reply:
oh i see now, sure that's fine